streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -589,11 +589,7 @@ boolean runLoop() {
             } catch (final TaskCorruptedException e) {
                 log.warn("Detected the states of tasks " + e.corruptedTasks() + " are corrupted. " +
                         "Will close the task as dirty and re-create and bootstrap from scratch.", e);
-                try {
-                    taskManager.handleCorruption(e.corruptedTasks());
-                } catch (final TaskMigratedException taskMigrated) {
```

Comment:
Now that we've removed the commit, there's no reason for `handleCorruption` to ever throw TaskMigratedException. Bonus cleanup! 😄 

(All we do now in `closeAndRevive` is flush the stores and force a checkpoint with corrupted changelogs removed, suspend -> close -> revive the task, and pause the consumer -- no way to hit TaskMigrated)